### ISSUE TITLE
Only prompt for sudo if capabilities to install (requires zypper change)

### ISSUE THREAD

**Issue by boombatower** [steamtricks/steamtricks#44]
As the comment in the code indicates this was something I would like to see fixed in zypper. I filed an issue openSUSE/zypper#115 to hope someone will get to it before me. This issue is a placeholder to utilize the change in steamtricks once it is present in zypper.

**Comment by mlandres** [comment]
What about `zypper se -i --provides <some_capability>` ?

**Comment by boombatower** [comment]
`s/--exact/-x/`

Seems like it is close although I would either need to check each capability individually to see that at least one package is installed (cumbersome and slow) or have some way to differentiate from extra providers vs ones that need to be installed.

As an example:
```
$ zypper se -x --provides "libstdc++.so.6(CXXABI_1.3)(64bit)"
S | Name            | Summary                         | Type   
--+-----------------+---------------------------------+--------
i | libstdc++6      | The standard C++ shared library | package
  | libstdc++6-gcc6 | The standard C++ shared library | package
```

vs

```
$ sudo zypper in -C "libstdc++.so.6(CXXABI_1.3)(64bit)"
'libstdc++6' providing 'libstdc++.so.6(CXXABI_1.3)(64bit)' is already installed.
```

Is there a way in which I can differentiate?

**Comment by boombatower** [comment]
I overlooked your intent regarding the `-i` suggestion. That said it would not appear that a set of capabilities can be tried all at once since `search` treats that as an `and` rather than `install` which treats looks for any package to provide each and removes duplicates. As such this approach would require that each one be tried individually and without `-i` to see if there are providers in the case that not installed.

**Comment by mlandres** [comment]
You live and learn. Looks like zypper does not perform the `uid==0` check if you explicitly use `--root`.

In this case it all depends on whether you have sufficient permissions and it also works with `--root /`.
So maybe `zypper --root / --no-refresh in --dry-run -C ...`?

**Comment by boombatower** [comment]
Now that is some magic! Presumably this is a stable enough behavior on which one could depend? Do you see the feature of install anything missing in for these binaries as being something acceptable/desirable for addition to `zypper` itself?

**Comment by mlandres** [comment]
The `--root` behavior is intended. The uid check is just to prevent you from accidentally starting an action you can not complete.
Regarding the feature, I don't think this is something for a zypper built-in command.

**Comment by boombatower** [comment]
Fixed in 2f59493c082c89e319d1d6c8a0898653adad3ca5. Thanks!
